README.rst
Python Client for AI Platform Notebooks API
===========================================

|stable| |pypi| |versions|

`AI Platform Notebooks API`_: 

- `Client Library Documentation`_
- `Product Documentation`_

.. |stable| image:: https://img.shields.io/badge/support-stable-gold.svg
   :target: https://github.com/googleapis/google-cloud-python/blob/main/README.rst#stability-levels
.. |pypi| image:: https://img.shields.io/pypi/v/google-cloud-notebooks.svg
   :target: https://pypi.org/project/google-cloud-notebooks/
.. |versions| image:: https://img.shields.io/pypi/pyversions/google-cloud-notebooks.svg
   :target: https://pypi.org/project/google-cloud-notebooks/
.. _AI Platform Notebooks API: https://cloud.google.com/ai-platform/notebooks/
.. _Client Library Documentation: https://cloud.google.com/python/docs/reference/notebooks/latest
.. _Product Documentation:  https://cloud.google.com/ai-platform/notebooks/

Quick Start
-----------

In order to use this library, you first need to go through the following steps:

1. `Select or create a Cloud Platform project.`_
2. `Enable billing for your project.`_
3. `Enable the AI Platform Notebooks API.`_
4. `Setup Authentication.`_

.. _Select or create a Cloud Platform project.: https://console.cloud.google.com/project
.. _Enable billing for your project.: https://cloud.google.com/billing/docs/how-to/modify-project#enable_billing_for_a_project
.. _Enable the AI Platform Notebooks API.:  https://cloud.google.com/ai-platform/notebooks/
.. _Setup Authentication.: https://googleapis.dev/python/google-api-core/latest/auth.html

Installation
~~~~~~~~~~~~

Install this library in a `virtualenv`_ using pip. `virtualenv`_ is a tool to
create isolated Python environments. The basic problem it addresses is one of
dependencies and versions, and indirectly permissions.

With `virtualenv`_, it's possible to install this library without needing system
install permissions, and without clashing with the installed system
dependencies.

.. _`virtualenv`: https://virtualenv.pypa.io/en/latest/


Code samples and snippets
~~~~~~~~~~~~~~~~~~~~~~~~~

Code samples and snippets live in the `samples/` folder.


Supported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^
Our client libraries are compatible with all current [active](https://devguide.python.org/devcycle/#in-development-main-branch) and [maintenance](https://devguide.python.org/devcycle/#maintenance-branches) versions of
Python.

Python >= 3.7

Unsupported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^^^
Python <= 3.6

If you are using an [end-of-life](https://devguide.python.org/devcycle/#end-of-life-branches)
version of Python, we recommend that you update as soon as possible to an actively supported version.


Mac/Linux
^^^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    source <your-env>/bin/activate
    <your-env>/bin/pip install google-cloud-notebooks


Windows
^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    <your-env>\Scripts\activate
    <your-env>\Scripts\pip.exe install google-cloud-notebooks

Next Steps
~~~~~~~~~~

-  Read the `Client Library Documentation`_ for AI Platform Notebooks API
   to see other available methods on the client.
-  Read the `AI Platform Notebooks API Product documentation`_ to learn
   more about the product and see How-to Guides.
-  View this `README`_ to see the full list of Cloud
   APIs that we cover.

.. _AI Platform Notebooks API Product documentation:  https://cloud.google.com/ai-platform/notebooks/
.. _README: https://github.com/googleapis/google-cloud-python/blob/main/README.rst


CHANGELOG.md
# Changelog

### [0.3.2](https://www.github.com/googleapis/python-notebooks/compare/v0.3.1...v0.3.2) (2021-07-28)


### Features

* add Samples section to CONTRIBUTING.rst ([#64](https://www.github.com/googleapis/python-notebooks/issues/64)) ([2e9c620](https://www.github.com/googleapis/python-notebooks/commit/2e9c620c45723fe8de13b437fb9642d14b2d8029))


### Bug Fixes

* enable self signed jwt for grpc ([#68](https://www.github.com/googleapis/python-notebooks/issues/68)) ([ec048cd](https://www.github.com/googleapis/python-notebooks/commit/ec048cd10d04b76fcbac5a246e3bcb6ef81353c8))


### Miscellaneous Chores

* release as 0.3.2 ([#69](https://www.github.com/googleapis/python-notebooks/issues/69)) ([e3eb4df](https://www.github.com/googleapis/python-notebooks/commit/e3eb4df74a19b88f4f60fab87b9d85349f025ce0))

### [0.3.1](https://www.github.com/googleapis/python-notebooks/compare/v0.3.0...v0.3.1) (2021-07-20)


### Bug Fixes

* **deps:** pin 'google-{api,cloud}-core', 'google-auth' to allow 2.x versions ([#63](https://www.github.com/googleapis/python-notebooks/issues/63)) ([dcfad3e](https://www.github.com/googleapis/python-notebooks/commit/dcfad3ed534fc8150d0f2ff349da78f22b72b6b3))

## [0.3.0](https://www.github.com/googleapis/python-notebooks/compare/v0.2.0...v0.3.0) (2021-07-01)


### Features

* add always_use_jwt_access ([#56](https://www.github.com/googleapis/python-notebooks/issues/56)) ([a60c86d](https://www.github.com/googleapis/python-notebooks/commit/a60c86d69ebe76897d63ee53b0a7aa44c3a32013))


### Bug Fixes

* disable always_use_jwt_access ([#60](https://www.github.com/googleapis/python-notebooks/issues/60)) ([336ab30](https://www.github.com/googleapis/python-notebooks/commit/336ab303d677415d4c3de172768f8236379c5f90))


### Documentation

* omit mention of Python 2.7 in 'CONTRIBUTING.rst' ([#1127](https://www.github.com/googleapis/python-notebooks/issues/1127)) ([#51](https://www.github.com/googleapis/python-notebooks/issues/51)) ([4b4cb00](https://www.github.com/googleapis/python-notebooks/commit/4b4cb00dc4245ab5090eb29434af30a3b4e736a0)), closes [#1126](https://www.github.com/googleapis/python-notebooks/issues/1126)

## [0.2.0](https://www.github.com/googleapis/python-notebooks/compare/v0.1.2...v0.2.0) (2021-05-28)


### Features

* add `from_service_account_info` ([#26](https://www.github.com/googleapis/python-notebooks/issues/26)) ([4999922](https://www.github.com/googleapis/python-notebooks/commit/4999922dc0f6eaebc8aec58929176ab6b87cfdca))
* support self-signed JWT flow for service accounts ([7a84b3b](https://www.github.com/googleapis/python-notebooks/commit/7a84b3b9b8c206a0dc33ccc09821ffa8ee8c3ddd))


### Bug Fixes

* add async client to %name_%version/init.py ([7a84b3b](https://www.github.com/googleapis/python-notebooks/commit/7a84b3b9b8c206a0dc33ccc09821ffa8ee8c3ddd))
* **deps:** add packaging requirement ([#45](https://www.github.com/googleapis/python-notebooks/issues/45)) ([9790dc9](https://www.github.com/googleapis/python-notebooks/commit/9790dc9da532ec396a8d81e3946da53cf243c066))

### [0.1.2](https://www.github.com/googleapis/python-notebooks/compare/v0.1.1...v0.1.2) (2021-02-08)


### Bug Fixes

* remove gRPC send/recv limit ([#12](https://www.github.com/googleapis/python-notebooks/issues/12)) ([8faa7fc](https://www.github.com/googleapis/python-notebooks/commit/8faa7fc87f62590b5c4119dc63d08347ed8eb901))

### [0.1.1](https://www.github.com/googleapis/python-notebooks/compare/v0.1.0...v0.1.1) (2020-08-06)


### Bug Fixes

* fix package name ([#2](https://www.github.com/googleapis/python-notebooks/issues/2)) ([ae80dcf](https://www.github.com/googleapis/python-notebooks/commit/ae80dcffc544a31096e9f076e9538985c1b3a44f))

## 0.1.0 (2020-08-04)


### Features

* generate v1 ([cb8049b](https://www.github.com/googleapis/python-notebooks/commit/cb8049bc35322565c6fd3f04955b756ba9a3415a))


owl-bot-staging/v1beta1/noxfile.py
# -*- coding: utf-8 -*-
# Copyright 2020 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
import os
import pathlib
import shutil
import subprocess
import sys


import nox  # type: ignore

CURRENT_DIRECTORY = pathlib.Path(__file__).parent.absolute()

LOWER_BOUND_CONSTRAINTS_FILE = CURRENT_DIRECTORY / "constraints.txt"
PACKAGE_NAME = subprocess.check_output([sys.executable, "setup.py", "--name"], encoding="utf-8")


nox.sessions = [
    "unit",
    "cover",
    "mypy",
    "check_lower_bounds"
    # exclude update_lower_bounds from default
    "docs",
]

@nox.session(python=['3.6', '3.7', '3.8', '3.9'])
def unit(session):
    """Run the unit test suite."""

    session.install('coverage', 'pytest', 'pytest-cov', 'asyncmock', 'pytest-asyncio')
    session.install('-e', '.')

    session.run(
        'py.test',
        '--quiet',
        '--cov=google/cloud/notebooks_v1beta1/',
        '--cov-config=.coveragerc',
        '--cov-report=term',
        '--cov-report=html',
        os.path.join('tests', 'unit', ''.join(session.posargs))
    )


@nox.session(python='3.7')
def cover(session):
    """Run the final coverage report.
    This outputs the coverage report aggregating coverage from the unit
    test runs (not system test runs), and then erases coverage data.
    """
    session.install("coverage", "pytest-cov")
    session.run("coverage", "report", "--show-missing", "--fail-under=100")

    session.run("coverage", "erase")


@nox.session(python=['3.6', '3.7'])
def mypy(session):
    """Run the type checker."""
    session.install('mypy')
    session.install('.')
    session.run(
        'mypy',
        '--explicit-package-bases',
        'google',
    )


@nox.session
def update_lower_bounds(session):
    """Update lower bounds in constraints.txt to match setup.py"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'update',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )


@nox.session
def check_lower_bounds(session):
    """Check lower bounds in setup.py are reflected in constraints file"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'check',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )

@nox.session(python='3.6')
def docs(session):
    """Build the docs for this library."""

    session.install("-e", ".")
    session.install("sphinx<3.0.0", "alabaster", "recommonmark")

    shutil.rmtree(os.path.join("docs", "_build"), ignore_errors=True)
    session.run(
        "sphinx-build",
        "-W",  # warnings as errors
        "-T",  # show full traceback on exception
        "-N",  # no colors
        "-b",
        "html",
        "-d",
        os.path.join("docs", "_build", "doctrees", ""),
        os.path.join("docs", ""),
        os.path.join("docs", "_build", "html", ""),
    )
